File: src/main/java/pharmacyCompany/model/Purchase.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pharmacyCompany.model;

import java.util.ArrayList;
import java.util.Objects;

/**
 *
 * @author Alumne
 */
public class Purchase implements Entity {

    private int id;
    private int idUser;
    private int idProduct;
    private String deliveryDate;
    private String specialRequests;
    private String specialInstructions;

    public Purchase() {

    }

    public Purchase(int id, int idUser, int idProduct, String deliveryDate, String specialRequests, String specialIntructions) {
        this.id = id;
        this.idUser = idUser;
        this.idProduct = idProduct;
        this.deliveryDate = deliveryDate;
        this.specialRequests = specialRequests;
        this.specialInstructions = specialIntructions;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    public int getIdProduct() {
        return idProduct;
    }

    public void setIdProduct(int idProduct) {
        this.idProduct = idProduct;
    }
    
    public String getDeliveryDate() {
        return deliveryDate;
    }
    
    public void setDeliveryDate(String deliveryDate) {
        this.deliveryDate = deliveryDate;
    }
    
    public String getSpecialRequests() {
        return specialRequests;
    }
    
    public void setSpecialRequests(String specialRequests) {
        this.specialRequests = specialRequests;
    }

    public String getSpecialInstructions() {
        return specialInstructions;
    }

    public void setSpecialInstructions(String specialInstructions) {
        this.specialInstructions = specialInstructions;
    }
    
    @Override
    public int hashCode() {
        int hash = 7;
        hash = 97 * hash + Objects.hashCode(this.id);
        return hash;
    }

    @Override
    public boolean equals(Object obj) {
        if (obj == null) {
            return false;
        }
        if (getClass() != obj.getClass()) {
            return false;
        }
        final Purchase other = (Purchase) obj;
        if (!Objects.equals(this.id, other.id)) {
            return false;
        }
        return true;
    }

    @Override
    public String toString() {
        return "Purchase{" + this.id + " idUser=" + this.idUser + ", idProduct=" + this.idProduct + ", deliveryDate=" + this.deliveryDate + "}";
    }
}

File: src/main/webapp/js/model/Purchase.js
function Purchase() {
    //Attributes declaration
    this.id;
    this.idUser;
    this.idProduct;
    this.deliveryDate;
    this.specialRequests = new Array();
    this.specialInstructions;

    //methods declaration
    this.construct = function (id, idUser, idProduct, deliveryDate, specialRequests, specialInstructions) {
        this.setId(id);
        this.setIdUser(idUser);
        this.setIdProduct(idProduct);
        this.setDeliveryDate(deliveryDate);
        this.setSpecialRequests(specialRequests);
        this.setSpecialInstructions(specialInstructions);
    }

    //setters
    this.setId = function (id) {
        this.id = id;
    }
    this.setIdUser = function (idUser) {
        this.idUser = idUser;
    }
    this.setIdProduct = function (idProduct) {
        this.idProduct = idProduct;
    }
    this.setDeliveryDate = function (deliveryDate) {
        this.deliveryDate = deliveryDate;
    }
    this.setSpecialRequests = function (specialRequests) {
        this.specialRequests = specialRequests;
    }
    this.setSpecialInstructions = function (specialInstructions) {
        this.specialInstructions = specialInstructions;
    }

    //getters
    this.getId = function () {
        return this.id;
    }
    this.getIdUser = function () {
        return this.idUser;
    }
    this.getIdProduct = function () {
        return this.idProduct;
    }
    this.getDeliveryDate = function () {
        return this.deliveryDate;
    }
    this.getSpecialRequests = function () {
        return this.specialRequests;
    }
    this.getSpecialInstructions = function () {
        return this.specialInstructions;
    }

    this.addSpecialRequests = function (specialreq) {
        this.specialRequests.push(specialreq);
    }

    this.removeSpecialRequests = function (specialreq) {
        for (var i = 0; i < this.getSpecialRequests().length; i++) {
            if (this.getSpecialRequests()[i] == specialreq) {
                this.specialRequests.splice(i, 1);
                break;
            }
        }
    }

    this.toString = function () {
        var purchaseString = "id=" + this.getId() + " idUser=" + this.getIdUser() + " idProduct=" + this.getIdProduct();
        purchaseString += " deliveryDate=" + this.getDeliveryDate() + " specialRequests=" + this.getSpecialRequests();
        purchaseString += " specialInstructions=" + this.getSpecialInstructions();

        return purchaseString;
    }
}
